allensdk/brain_observatory/behavior/stimulus_processing.py

```diff
@@ -559,3 +559,67 @@ def is_change_event(stimulus_presentations: pd.DataFrame) -> pd.Series:
     is_change = is_change.fillna(False)
 
     return is_change
+
+
+def compute_trials_id_for_stimulus(
+        stim_pres_table: pd.DataFrame,
+        trials_table: pd.DataFrame) -> pd.Series:
+    """Add an id to allow for merging of the stimulus presentations
+    table with the trials table.
+
+    Parameters
+    ----------
+    stim_pres_table : pandas.DataFrame
+        Pandas stimulus table to create trials_id from.
+    trials_table : pandas.DataFrame
+        Trials table to create id from using trial start times.
+
+    Returns
+    -------
+    trials_id : pd.Series
+        Unique id to allow merging of the stim table with the trials table.
+        Null values are represented by -1.
+    """
+    stim_pres_sorted = stim_pres_table.sort_values('start_time')
+    trials_sorted = trials_table.sort_values('start_time')
+    # Create a placeholder for the trials_id.
+    trials_ids = pd.Series(
+        data=np.full(len(stim_pres_sorted), -1, dtype=int),
+        index=stim_pres_sorted.index,
+        name='trials_id')
+
+    # Find stimulus blocks that start within a trial. Copy the trial_id
+    # into our new trials_ids series.
+    for idx, trial in trials_sorted.iterrows():
+        stim_mask = (stim_pres_sorted.start_time > trial.start_time) \
+                    & (stim_pres_sorted.start_time < trial.stop_time)
+        trials_ids[stim_mask] = idx
+
+    # The code below finds all stimulus blocks that contain images/trials
+    # and attempts to detect blocks that are identical to copy the associated
+    # trials_ids into those blocks. In the parlance of the data this is
+    # copying the active stimulus block data into the passive stimulus block.
+
+    # Get the block ids for the behavior trial presentations
+    stim_image_names = stim_pres_sorted.image_name
+    stim_blocks = stim_pres_sorted.stimulus_block
+    image_stim_blocks = stim_pres_sorted[
+        ~stim_image_names.isna()].stimulus_block.unique()
+
+    # Get the first block that has images associated with it.
+    # Assume it is the 'active' block.
+    first_block_id = image_stim_blocks[0]
```

Comment:
Sorry to be a pain here, but I think there might be a cleaner way to do this. In the code on 593 - 596 you are finding all of the active blocks. You can store a set of stimulus blocks there, which are guaranteed to be the active ones. Then, on 616 - 623, you can check whether any of the passive_stim_blocks are equal to any of the active_stim_blocks. This has the advantage of supporting multiple active stimulus blocks, if they ever design a session like that.

Reply:
Okay, so is the trick here that if stimulus start time is within a trial, that stimulus can be denoted as `active` so I could basically, ass this as a stimulus processing step if I wanted to.

Reply:
yes, I didn't think of that before thinking about this PR, but that should be true.

Reply:
Okay, do we want to use this function then to also add the `active` block tag if it doesn't already exist? That would simplify the code below as you suggest.

Reply:
Are you asking to add this as a column to the table? I can't remember -- is that something we have to do for some other reason? Go for it, if it makes sense to add it.